chain/processor.go

```diff
@@ -58,50 +56,27 @@ func (b *StatelessBlock) Execute(
 			e.Stop()
 			return nil, nil, err
 		}
+		// Fetch keys from disk or check if it's in cache
+		wg := f.Lookup(ctx, tx.ID(), stateKeys)
 		e.Run(stateKeys, func() error {
+			// Block until worker finishes fetching keys
+			wg.Wait()
```

Comment:
I think this would be a lot cleaner if `wg.Wait()` returned `reads` and `storage`